### PR TITLE
[Infer] Serving example w/ ray-serve (multiple GPU case)

### PR DESCRIPTION
**Pull request by yuanheng-zhao**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 📝 What does this PR do?

- This PR adds necessary files for Colossal Inference serving with Ray Serve. Steps of deployment were added into README.
- This demo is used for testing and demonstrating the usage of Colossal Inference from colossalai.inference.tensor_parallel with ray serve. Serving for multiple-gpu inference is supported (e.g. tensor parallel with colossalai).

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/kernel/triton/__init__.py [hunk 1]
```diff
@@ -21,7 +22,7 @@
         "copy_kv_cache_to_dest",
         "rotary_embedding_fwd",
         "token_attention_fwd",
-        "gptq_fused_linear_triton",
+        # "gptq_fused_linear_triton",
```
Comment (yuanheng-zhao) [inline colossalai/kernel/triton/__init__.py:25]: gptq imports issue: might want to fix in another PR.

File: examples/inference/serving/ray_serve/Colossal_Inference_rayserve.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        collective.init_collective_group(world_size, rank, "nccl", "default")
+        # initialize and set distributed environment
+        colossalai.launch(config={}, rank=rank, world_size=world_size, host="localhost", port=port, backend="nccl")
+        ray_serve_logger.info(f"Worker with rank {rank} (world size {world_size}) setting up..")
+        log_cuda_info("Worker.setup")
+
+        # Load model
+        self.tokenizer = BloomTokenizerFast.from_pretrained(self.model_path)
+        self.tokenizer.pad_token = self.tokenizer.eos_token
+        self.model = BloomForCausalLM.from_pretrained(
```
Comment (Xu-Kai) [inline examples/inference/serving/ray_serve/Colossal_Inference_rayserve.py:66]: AutoTokenizer and AutoModel are better.

Comment (yuanheng-zhao) [reply examples/inference/serving/ray_serve/Colossal_Inference_rayserve.py]: Gotcha. Revised.

**Review by Xu-Kai — APPROVED** [review]

File: colossalai/kernel/triton/__init__.py (hunk 1 above)
Comment (yuanheng-zhao) [reply colossalai/kernel/triton/__init__.py:25]: To be fixed in https://github.com/hpcaitech/ColossalAI/pull/4838

File: colossalai/inference/tensor_parallel/modeling/__init__.py [hunk 3]
```diff
@@ -1,5 +1,3 @@
-import _utils
```
Comment (yuanheng-zhao) [inline colossalai/inference/tensor_parallel/modeling/__init__.py:1]: import issue: preferably to be fixed in https://github.com/hpcaitech/ColossalAI/pull/4830

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**